lib/response.js

```diff
@@ -20,7 +20,27 @@ function Response (req, onEnd) {
   this.once('finish', () => {
     const res = generatePayload(this)
     res.raw.req = req
-    process.nextTick(() => onEnd(res))
+    if (typeof reject === 'function') {
+      process.nextTick(() => onEnd(res))
+    } else {
+      process.nextTick(() => onEnd(null, res))
+    }
+  })
+
+  this.connection.once('error', err => {
```

Comment:
Connection issues should be covered by a test too.

Reply:
Connection errors are covered by
https://github.com/fastify/light-my-request/blob/ad9cc6e87df1235e92ffbd8ec38853245195386c/test/test.js#L757-L777

The res error has not test at the moment.